### PR TITLE
update access files

### PR DESCRIPTION
**Pull request by TomerGoldfeder**
box, dashboard, product

### CODE REVIEW

File: src/main/me/wapy/controllers/TrafficController.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import me.wapy.utils.JSONResponse;
+import me.wapy.utils.RESTRoute;
+import net.sf.json.JSON;
+import spark.Request;
+import spark.Response;
+
+import java.sql.Time;
+import java.sql.Timestamp;
+
+public class TrafficController implements RESTRoute {
```
Comment (Minitour) [inline src/main/me/wapy/controllers/TrafficController.java:15]: The traffic data should not be returned separately. The controller name should indeed be `DashboardController`. 
When the controller is called you will need to create an instance of `DashboardAccess` and then run the relevant queries.

The results from the query should be stored in a JSON object.

Please lookup how to use Gson. a JSON parsing/handling library by google. Note that you don't have to create corresponding classes for each response, and can construct a JSON object on the fly using the `addProperty` method.

File: src/main/me/wapy/database/data_access/BoxAccess.java [hunk 2]
```diff
@@ -30,16 +33,16 @@ public BoxAccess(Database database) {
      * @param toTime
      * @return
      */
-    public List<Product> getAllProductsInWindow(String cameraId, Timestamp fromTime, Timestamp toTime) throws SQLException {
+    public List<Product> getAllProductsInWindow(AuthContext context, String cameraId, Timestamp fromTime, Timestamp toTime) throws SQLException {
```
Comment (Minitour) [inline src/main/me/wapy/database/data_access/BoxAccess.java:36]: There is no need for AuthContext in this application. Authentication will happen behind the scenes using JWT. You can assume that the API call has already been authenticated.

File: src/main/me/wapy/database/data_access/BoxAccess.java [hunk 3]
```diff
@@ -68,7 +71,7 @@ public Product getMostViewedProductInWindow(String cameraId, Timestamp fromTime,
         );
 
         if (res.isEmpty())
-            return null;
+            return new Product();
```
Comment (Minitour) [inline src/main/me/wapy/database/data_access/BoxAccess.java:74]: In case we return an single object and the object does not exist we shall return null, in the case of lists we shall return an empty list. 

The reason we do that is because sometimes we search for multiple objects that apply to certain criterias. and in the case none were found an empty list is expected rather than null.

However when we search for a single object, and that object is not found, it simply means it does not exist, therefor we expect either a value or a null.

**TL;DR**
- For objects: Value or Null
- For lists: List or empty list

File: src/main/me/wapy/database/data_access/BoxAccess.java [hunk 4]
```diff
@@ -114,4 +117,49 @@ public Product getLeastViewedProductInWindow(String cameraId, Timestamp fromTime
...
+    /**
+     * Return all reactions for specific product and box in a given time interval
+     * @param objectId
+     * @param cameraId
+     * @param fromTime
+     * @param toTime
+     * @return
+     * @throws SQLException
+     */
+    public List<Reaction> getAllReactionsPerProductPerBox(AuthContext context, String objectId, String cameraId, Timestamp fromTime, Timestamp toTime) throws SQLException {
```
Comment (Minitour) [inline src/main/me/wapy/database/data_access/BoxAccess.java:129]: I will take care of this method.

**Review by Minitour — COMMENTED** [review]
There is room for improvement but overall great job.

File: src/main/me/wapy/database/data_access/BoxAccess.java (hunk 4 above)
Comment (TomerGoldfeder) [reply src/main/me/wapy/database/data_access/BoxAccess.java:129]: ok

File: src/main/me/wapy/database/data_access/BoxAccess.java (hunk 3 above)
Comment (TomerGoldfeder) [reply src/main/me/wapy/database/data_access/BoxAccess.java:74]: changed to expected values

File: src/main/me/wapy/database/data_access/BoxAccess.java (hunk 2 above)
Comment (TomerGoldfeder) [reply src/main/me/wapy/database/data_access/BoxAccess.java:36]: removed auth

File: src/main/me/wapy/controllers/TrafficController.java (hunk 1 above)
Comment (TomerGoldfeder) [reply src/main/me/wapy/controllers/TrafficController.java:15]: will implement and start a new pull request

**Comment by TomerGoldfeder** [comment]
will do the changes and start a new PR
